gorder.go

```diff
@@ -1,29 +1,24 @@
 package gorder
+import "errors"
 
-import (
-	"errors"
-	"regexp"
-)
+type Algo int
 
-func TopologicalSort(digraph map[interface{}][]interface{}, algorithm string) (solution []interface{}, err error) {
-	kahnRgxp, err := regexp.Compile(`[Kk]ahn\z`)
-	if err != nil {
-		return nil, err
-	}
-	dfsBasedRgxp, err := regexp.Compile(`[Dd][Ff][Ss]-?[Bb]ased\z`)
-	if err != nil {
-		return nil, err
-	}
+const (
+	DFS Algo = iota
+	Kahn
+)
```

Comment:
I doubt anyone's heavily using this package, but I firmly suggest being backwards-compatible here. Like this:

```suggestion
const (
	DFS  Algo = "dfs-based"
	Kahn Algo = "kahn"
)
```

This way, we won't break anything. Reference: https://play.golang.org/p/EHDuPFPqgqs